File: src/test/java/runtime/judge/TestJudge.java
package runtime.judge;

import org.junit.jupiter.api.Test;
import runtime.parser.Example;

import java.lang.reflect.Method;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class TestJudge {

	public int sample(int a, int b) { return a + b; }

	@Test
	public void test() throws ReflectiveOperationException {
		Method method = TestJudge.class.getMethod("sample", int.class, int.class);
		Example example_1 = new Example("1, 1", "2"); // pass
		Example example_2 = new Example("2, 2", "3"); // wrong answer

		List<Judge.Result> results = Judge.run(method, List.of(example_1, example_2));
		assertThat(results).hasSize(2);
		assertThat(results.get(0).pass()).isTrue();
		assertThat(results.get(1).pass()).isFalse();
	}

}

File: src/test/scala/runtime/judge/test_judge.scala
package runtime.judge

import org.scalatest.funsuite.AnyFunSuite
import runtime.parser.Example
import runtime.repo.{MetaDataNode, MetaDataParamNode, MetaDataReturnNode}

class test_judge extends AnyFunSuite {

	object Sample {
		def sample_add(a: Int, b: Int): Int = a + b
	}

	test("sample") {
		val method = Sample.getClass.getDeclaredMethod("sample_add", classOf[Int], classOf[Int])
		val meta = MetaDataNode("sample_add", Array(
			MetaDataParamNode("a", "integer"), MetaDataParamNode("b", "integer")
		), MetaDataReturnNode("integer"), manual = false, output = null)

		val example_1 = Example("test 1", "1,2", Some("3")) // ok
		val result_1 = Judge.run(Sample, method, example_1, meta)
		assert(result_1.ok)

		val example_2 = Example("test 2", "1,2", Some("4")) // wrong answer
		val result_2 = Judge.run(Sample, method, example_2, meta)
		assert(!result_2.ok)
	}

}
